### ISSUE TITLE
3.4.3 introduced bug in enforce-consistent-class-order

### ISSUE THREAD

**Issue by donaldpipowitch** [schoero/eslint-plugin-better-tailwindcss#150]
Since 3.4.3 I see some weird error reporting in my IDE (Cursor 1.2.1) that I don't see in the CLI or when I go back to 3.4.2. I assume this was introduced by https://github.com/schoero/eslint-plugin-better-tailwindcss/pull/146.

What happens? In my IDE this is reported as an error (fine via CLI or 3.4.2):

```
            className={cn(
              `
                mask-no-repeat
                max-[560px]:relative max-[560px]:mask-l-from-[20%]
                max-[560px]:mask-l-to-130%
                max-[560px]:before:pointer-events-none
                max-[560px]:before:absolute max-[560px]:before:top-0
                max-[560px]:before:left-[0] max-[560px]:before:z-3
                max-[560px]:before:hidden max-[560px]:before:h-full
                max-[560px]:before:w-[60%] max-[560px]:before:bg-gradient-to-r
                max-[560px]:before:from-background-sidebar-default
                max-[560px]:before:to-transparent max-[560px]:before:opacity-70
              `,
              activeSidebar === 'contextual' &&
                `max-[560px]:mask-none max-[560px]:before:hidden`
            )}
```

It gets auto-fixed to this:

```
            className={cn(
              `
                mask-no-repeat
                max-[560px]:relative max-[560px]:mask-l-from-[20%]
                max-[560px]:mask-l-to-130%
                max-[560px]:before:pointer-events-none
                max-[560px]:before:absolute max-[560px]:before:top-0
                max-[560px]:before:left-[0] max-[560px]:before:z-3
                max-[560px]:before:h-full max-[560px]:before:w-[60%]
                max-[560px]:before:bg-gradient-to-r
                max-[560px]:before:from-background-sidebar-default
                max-[560px]:before:to-transparent max-[560px]:before:opacity-70
                max-[560px]:before:hidden
              `,
              activeSidebar === 'contextual' &&
                `max-[560px]:mask-none max-[560px]:before:hidden`
            )}
````

Interestingly this is still reported as an error. If I auto-fix it now it goes back to the initial variant. Endless loop.

**Note**: If I move this code to somewhere else it isn't reported as an error anymore, so I cannot create a reproduceable case. Also if I slightly change the case (like removing the part with `activeSidebar`) it disappears. As if this is somehow related to the file size/content hash...
It's the 2nd time I see this now since 3.4.3 was released, so it is also not limited to this particular file, but I fear it could randomly appear in any file.

**Note 2**: If I undo this change locally the issue disappears.
https://github.com/schoero/eslint-plugin-better-tailwindcss/pull/146/files#diff-d5ecd48e1fc9b3dd6eb6cd965af9c0ceb9e56c6a7cd70bda84fe8683c761404eR206-R211

**Note 3**: Maybe `classVariants` already creates wrong results. In one run I get `['max-[560px]:before:pointer-events-none',[ 'before', 'max-[560px]' ]]` and in the next run `['max-[560px]:before:pointer-events-none',[ 'max-[560px]','before' ]]`.

**Comment by schoero** [comment]
This took way too long for me to figure out 😅

tailwindcss internally caches the results of the `parseCandidate()` function. With the reverse function I altered the cache and on every call I reversed the cached order.

https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/0bdd65796c054f09cbd1362f3a7977bf0400ad3c/src/tailwindcss/class-variants.async.v4.ts#L11-L17

Thanks for the detailed report 👍
